substrate/frame/referenda/src/types.rs

```diff
@@ -118,10 +118,40 @@ pub struct Deposit<AccountId, Balance> {
 
 pub const DEFAULT_MAX_TRACK_NAME_LEN: usize = 25;
 
+struct BoundedSize<const N: usize>;
+
+impl<const N: usize> Get<u32> for BoundedSize<N> {
+	fn get() -> u32 {
+		N as u32
+	}
+}
+
+#[derive(Encode, Decode)]
+struct ArrayLikeVec<const N: usize>(BoundedVec<u8, BoundedSize<N>>);
+
+impl<'a, const N: usize> From<&'a [u8; N]> for ArrayLikeVec<N> {
+	fn from(bytes: &'a [u8; N]) -> Self {
+		ArrayLikeVec(BoundedVec::truncate_from(bytes.to_vec()))
+	}
+}
+
+impl<const N: usize> From<ArrayLikeVec<N>> for [u8; N] {
+	fn from(s: ArrayLikeVec<N>) -> Self {
+		let mut bytes = [0u8; N];
+		bytes[..N].copy_from_slice(&s.0[..N]);
+		bytes
+	}
+}
+
+impl<'a, const N: usize> codec::EncodeAsRef<'a, [u8; N]> for ArrayLikeVec<N> {
+	type RefType = ArrayLikeVec<N>;
+}
+
 /// Detailed information about the configuration of a referenda track
 #[derive(Clone, Encode, Decode, MaxEncodedLen, TypeInfo, Eq, PartialEq, Debug)]
 pub struct TrackInfo<Balance, Moment, const N: usize = DEFAULT_MAX_TRACK_NAME_LEN> {
 	/// Name of this track.
+	#[codec(encoded_as = "ArrayLikeVec<N>")]
 	pub name: [u8; N],
```

Comment:
Why was this changed to an array and not a static Str as before?

Reply:
it is to be able to decode it from storage, the intention is to be able to have tracks defined in storage.

Reply:
Yes. That's the reason.

@gui1117 I changed the approach to use `BoundedVec` as it encodes better.